### PR TITLE
Moves ALB alarms into ALB module

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Move ALB alarms into ALB module so all ALBs get alarms by default.

### Who is this change for?

Developers who like alarms.

### Have the following been considered/are they needed?

- [ ] Run `terraform apply`.

### CODE REVIEW

File: terraform/ecs_alb/alarms.tf [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  period              = "60"
+  statistic           = "Sum"
+  threshold           = "1"
+
+  dimensions {
+    LoadBalancer = "${aws_alb.ecs_service.arn_suffix}"
+  }
+
+  alarm_description = "This metric monitors the ${var.name} ALB _target_ for 400s (API only)"
+  alarm_actions     = []
```
Comment (alicefuzier) [inline terraform/ecs_alb/alarms.tf]: none of these alarms have actions, so I don't think anything will get sent to `alarm_topic_arn`, will it?

File: terraform/alb.tf [hunk 2]
```diff
@@ -10,6 +10,8 @@ module "services_alb" {
 
   certificate_domain = "services.wellcomecollection.org"
   vpc_id             = "${module.vpc_services.vpc_id}"
+
```
Comment (alexwlchan) [inline terraform/alb.tf]: any reason for the empty line?

File: terraform/lambdas.tf [hunk 3]
```diff
@@ -224,3 +224,10 @@ module "trigger_post_to_slack_esg_not_terminating" {
   lambda_function_arn  = "${module.lambda_post_to_slack.arn}"
   sns_trigger_arn      = "${module.ec2_instance_terminating_for_too_long_alarm.arn}"
 }
+
+module "trigger_post_to_slack_alb" {
+  source               = "./lambda/trigger_sns"
+  lambda_function_name = "${module.lambda_post_to_slack.function_name}"
+  lambda_function_arn  = "${module.lambda_post_to_slack.arn}"
+  sns_trigger_arn      = "${module.alb_alarm.arn}"
+}
```
Comment (alexwlchan) [inline terraform/lambdas.tf]: needs a `terraform fmt`, I think

File: terraform/ecs_alb/alarms.tf [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  namespace           = "AWS/ApplicationELB"
+  period              = "60"
+  statistic           = "Sum"
+  threshold           = "1"
+
+  dimensions {
+    LoadBalancer = "${aws_alb.ecs_service.arn_suffix}"
+  }
+
+  alarm_description = "This metric monitors the ${var.name} ALB for 500s (Loris & API)"
```
Comment (alexwlchan) [inline terraform/ecs_alb/alarms.tf]: Now that we're in the module, are these descriptions still accurate?

File: terraform/ecs_alb/alarms.tf [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  alarm_actions     = []
+}
```
Comment (alexwlchan) [inline terraform/ecs_alb/alarms.tf]: Maybe these could be forced down into a further module? Something like:

```hcl
module "alb_500_errors" {
  source        = "./alb_alarm"
  alarm_name    = "${var.name}-alb-500-errors"
  metric_name   = "HTTPCode_ELB_5XX_Count"
  load_balancer = "${aws_alb.ecs_service.arn_suffix}"
  description   = "This metric monitors the ${var.name} ALB for 500s"
  alarm_actions = []
}
```

File: terraform/ecs_alb/variables.tf [hunk 6]
```diff
@@ -23,3 +23,7 @@ variable "health_check_path" {
 variable "vpc_id" {
   description = "ID of VPC to create ALB in"
 }
+
+variable "alarm_topic_arn" {
```
Comment (alicefuzier) [inline terraform/ecs_alb/variables.tf]: Would it be worth having two different topics for alarms? I don't think we will treat 500 errors the same way as 400 errors

File: terraform/alb.tf (hunk 2 above)
Comment (kenoir) [inline terraform/alb.tf]: ease of reading

File: terraform/lambdas.tf [hunk 7]
```diff
@@ -224,3 +224,17 @@ module "trigger_post_to_slack_esg_not_terminating" {
...
 }
+
+module "trigger_post_to_slack_server_error_alb" {
+  source               = "./lambda/trigger_sns"
+  lambda_function_name = "${module.lambda_post_to_slack.function_name}"
+  lambda_function_arn  = "${module.lambda_post_to_slack.arn}"
+  sns_trigger_arn      = "${module.alb_server_error_alarm.arn}"
+}
+
+module "trigger_post_to_slack_client_error_alb" {
```
Comment (alicefuzier) [inline terraform/lambdas.tf]: do we want a slack notification for every 4xx?

Comment (kenoir) [inline terraform/lambdas.tf]: It's only a notification when it goes into ALARM state. I think right now we do (we shouldn't be seeing any really) - we can dial it back when the API is under more load.

**Pull request merged**